File: plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        bool hasFirst = 1;
        for(int i=0;i<digits.size();i++)
            if(digits[i] != 9)
            {
                hasFirst = 0;
                break;
            }
        vector<int> ans;
        if(hasFirst)
        {
            ans.push_back(1);
            for(int i=0;i<digits.size();i++)
                ans.push_back(0);
        }else
        {
            for(int i=digits.size()-1;i>-1;i--)
            {
                if(digits[i] == 9)
                    digits[i] = 0;
                else
                {
                    digits[i]++;
                    break;
                }
            }
            for(int i=0;i<digits.size();i++)
                ans.push_back(digits[i]);
        }
        return ans;
    }
};

File: July LeetCoding Challenge/Plus One.cs
public class Solution {
    public int[] PlusOne(int[] digits) {
        int n = digits.Length;
        for(int i=n-1;i>=0;i--)
        {
            if(digits[i] + 1 == 10)
                digits[i] = 0;
            else
            {
                digits[i]++;
                return digits;
            }
        }
        int[] res = new int[n+1];
        Array.Fill(res,0);
        res[0] = 1;
        return res;
    }
}
